pipeline/authn/authenticator_oauth2_client_credentials.go

```diff
@@ -110,37 +213,47 @@ func (a *AuthenticatorOAuth2ClientCredentials) Authenticate(r *http.Request, ses
 		return errors.Wrapf(helper.ErrUnauthorized, err.Error())
 	}
 
-	c := &clientcredentials.Config{
+	c := clientcredentials.Config{
 		ClientID:     user,
 		ClientSecret: password,
 		Scopes:       cf.Scopes,
 		TokenURL:     cf.TokenURL,
 		AuthStyle:    oauth2.AuthStyleInHeader,
 	}
 
-	token, err := c.Token(context.WithValue(
-		r.Context(),
-		oauth2.HTTPClient,
-		c.Client,
-	))
+	token := a.tokenFromCache(cf, c)
 
-	if err != nil {
-		if rErr, ok := err.(*oauth2.RetrieveError); ok {
-			switch httpStatusCode := rErr.Response.StatusCode; httpStatusCode {
-			case http.StatusServiceUnavailable:
+	// If the token can not be found, and the scope strategy is nil, and the required scope list
+	// is not empty, then we can not use the cache.
```

Comment:
The comment talks about scope list and scope strategy, but it does not look as if those values are being checked. Is this an old comment or is something missing? And what is the rationale behind not using the case in this scenario?

Reply:
This comment is a left over from the copied code in `authenticator_oauth2_introspection.go`, I'll remove it as it makes no sense here